src/ProgressOnderwijsUtils/Data/CascadedDelete.cs

```diff
@@ -13,6 +14,105 @@ namespace ProgressOnderwijsUtils
+            bool outputAllDeletedRows,
+            [CanBeNull] Action<string> logger,
+            [NotNull] string pkColumn,
+            [NotNull] params TId[] pksToDelete
+            )
+            where TId : Enum
+        {
+            var pkColumnSql = ParameterizedSql.CreateDynamic(pkColumn);
+            return RecursivelyDelete(conn, initialTableAsEntered, outputAllDeletedRows, logger, new[] { pkColumn }, SQL($@"
+                select {pkColumnSql} = q.QueryTableValue 
+                from {pksToDelete} q
+            "));
+        }
+
+        [NotNull]
+        public static DeletionReport[] RecursivelyDelete<TId>(
+            [NotNull] SqlCommandCreationContext conn,
+            ParameterizedSql initialTableAsEntered,
+            bool outputAllDeletedRows,
+            [CanBeNull] Action<string> logger,
+            [NotNull] params TId[] pksToDelete
+            )
+            where TId : IPropertiesAreUsedImplicitly, IMetaObject
+        {
+            var pksTable = SQL($"#pksTable");
+            var pkColumns = MetaObject.GetMetaProperties<TId>().Select(mp => mp.Name).ToArray();
+            var pkColumnsSql = pkColumns.Select(ParameterizedSql.CreateDynamic).ConcatenateSql(SQL($", "));
+
+            //union all is a nasty hack to enforce that the identity property is not propagated to the temp table
+            SQL($@"
+                select {pkColumnsSql} 
+                into {pksTable}
+                from {initialTableAsEntered}
+                where 1=0
+                union all
+                select {pkColumnsSql} 
+                from {initialTableAsEntered}
+                where 1=0
+            ").ExecuteNonQuery(conn);
+            pksToDelete.BulkCopyToSqlServer(conn, pksTable.CommandText());
+
+            var report = RecursivelyDelete(conn, initialTableAsEntered, outputAllDeletedRows, logger, pkColumns, SQL($@"
+                select {pkColumnsSql}
+                from {pksTable}
+            "));
+
+            SQL($@"
+                drop table {pksTable}
+            ").ExecuteNonQuery(conn);
+
+            return report;
+        }
+
+        [NotNull]
+        public static DeletionReport[] RecursivelyDelete(
+            [NotNull] SqlCommandCreationContext conn,
+            ParameterizedSql initialTableAsEntered,
+            bool outputAllDeletedRows,
+            [CanBeNull] Action<string> logger,
+            [NotNull] DataTable pksToDelete
+            )
+        {
+            var pksTable = SQL($"#pksTable");
+            var pkColumns = pksToDelete.Columns.Cast<DataColumn>().Select(dc => dc.ColumnName).ToArray();
+            var pkColumnsSql = pkColumns.Select(ParameterizedSql.CreateDynamic).ConcatenateSql(SQL($", "));
+
+            //union all is a nasty hack to enforce that the identity property is not propagated to the temp table
+            SQL($@"
+                select {pkColumnsSql} 
+                into {pksTable}
+                from {initialTableAsEntered}

```

Comment:
Deze DataTable en de IMetaObject versie maken een pksTable aan die nooit ids bevat dit niet echt in de tabel staan, maar de enum versie doet dat niet.

Maakt dat uit?

Reply:
laat maar